src/protections/HomeserverUserPolicyApplication/HomeserverUserPolicyApplication.ts

```diff
@@ -0,0 +1,204 @@
+// SPDX-FileCopyrightText: 2025 Gnuxie <Gnuxie@protonmail.com>
+//
+// SPDX-License-Identifier: AFL-3.0
+
+import { isError, Ok, Result } from "@gnuxie/typescript-result";
+import {
+  isStringUserID,
+  MatrixGlob,
+  StringServerName,
+  StringUserID,
+  userServerName,
+} from "@the-draupnir-project/matrix-basic-types";
+import {
+  LiteralPolicyRule,
+  Logger,
+  PolicyRuleChange,
+  PolicyRuleChangeType,
+  PolicyRuleMatchType,
+  PolicyRuleType,
+  Recommendation,
+  Task,
+  WatchedPolicyRooms,
+} from "matrix-protection-suite";
+import { UserRestrictionCapability } from "./UserRestrictionCapability";
+import { UserAuditLog } from "./UserAuditLog";
+
+const log = new Logger("HomeserverUserPolicyApplication");
+
+export class HomeserverUserPolicyApplication {
+  public constructor(
+    private readonly consequences: UserRestrictionCapability,
+    private readonly userAuditLog: UserAuditLog,
+    private readonly watchedPolicyRooms: WatchedPolicyRooms,
+    private readonly serverName: StringServerName,
+    private readonly automaticallyRedactForReasons: MatrixGlob[]
+  ) {
+    // nothing to do.
+  }
+
+  private isPolicyElegiableForRestriction(policy: LiteralPolicyRule): boolean {
+    if (policy.kind !== PolicyRuleType.User) {
+      return false;
+    }
+    if (!isStringUserID(policy.entity)) {
+      return false;
+    }
+    if (userServerName(policy.entity) !== this.serverName) {
+      return false;
+    }
+    if (policy.recommendation === Recommendation.Takedown) {
+      return true;
+    } else if (
+      // FIXME: This would be a policy eligible for purging,
+      // not suspension, any banned local user should be suspended.
+      policy.recommendation === Recommendation.Ban &&
+      this.automaticallyRedactForReasons.some((glob) =>
+        glob.test(policy.reason)
```

Comment:
We need to add some kind of prompt for using the deactivate command with the `purge-messages` option. And then make this suspend in any case.

Reply:
we did this